Dockerfiles/agent-ot/Dockerfile.byoc

```diff
@@ -0,0 +1,82 @@
+# Use the Ubuntu Slim AMD64 base image
+FROM ubuntu:24.04 AS builder
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV GO_VERSION=1.22.0
+ENV AGENT_VERSION=7.56.0-rc.7
+
+# Set the working directory
+WORKDIR /workspace
+
+# Update and install necessary packages
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    curl \
+    ca-certificates \
+    software-properties-common \
+    build-essential \
+    git \
+    python3 \
+    python3-pip \
+    python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Go based on architecture
+RUN ARCH=$(dpkg --print-architecture) && \
+    if [ "$ARCH" = "amd64" ]; then \
+    GO_ARCH="linux-amd64"; \
+    elif [ "$ARCH" = "arm64" ]; then \
+    GO_ARCH="linux-arm64"; \
+    else \
+    echo "Unsupported architecture: $ARCH" && exit 1; \
+    fi && \
+    curl -OL https://golang.org/dl/go${GO_VERSION}.$GO_ARCH.tar.gz && \
+    tar -C /usr/local -xzf go${GO_VERSION}.$GO_ARCH.tar.gz && \
+    rm go${GO_VERSION}.$GO_ARCH.tar.gz
+
+# Set up Go environment
+ENV PATH="/usr/local/go/bin:${PATH}"
+RUN mkdir /go
+ENV GOPATH=/go
+
+# Verify installations
+RUN go version && \
+    python3 --version && \
+    curl --version
+
+# Download and unpack source code
+#RUN curl -L https://github.com/DataDog/datadog-agent/archive/refs/tags/${AGENT_VERSION}.tar.gz -o datadog-agent-${AGENT_VERSION}.tar.gz && \
+#    tar -xzvf datadog-agent-${AGENT_VERSION}.tar.gz && \
+#    rm datadog-agent-${AGENT_VERSION}.tar.gz
+
+# TEMP: Use github source code
+RUN git clone https://github.com/DataDog/datadog-agent.git datadog-agent-${AGENT_VERSION}
+
+
+# Set the working directory to the source code
+WORKDIR /workspace/datadog-agent-${AGENT_VERSION}
+
+# Create and activate virtual environment, then install requirements
+RUN python3 -m venv venv && \
+    . venv/bin/activate && \
+    pip install --upgrade pip && \
+    pip install -r tasks/requirements.txt
+
+# Copy the manifest file
+COPY manifest.yaml /workspace/datadog-agent-${AGENT_VERSION}/comp/otelcol/collector-contrib/impl/manifest.yaml
+
+# Generate the files
+RUN . venv/bin/activate && invoke collector.generate
+
+# Build the OTel agent
+RUN . venv/bin/activate && invoke otel-agent.build
+
+# Use the final Datadog agent image
+FROM datadog/agent-dev:nightly-ot-beta-main-jmx
```

Comment:
I would make the tag here `nightly-ot-beta-main-jmx` also a build argument.

Reply:
done